public/app/plugins/datasource/influxdb-ifql/datasource.ts

```diff
@@ -0,0 +1,255 @@
+import _ from 'lodash';
+
+import * as dateMath from 'app/core/utils/datemath';
+
+import { getTableModelFromResult, getTimeSeriesFromResult, parseResults } from './response_parser';
+
+function serializeParams(params) {
+  if (!params) {
+    return '';
+  }
+
+  return _.reduce(
+    params,
+    (memo, value, key) => {
+      if (value === null || value === undefined) {
+        return memo;
+      }
+      memo.push(encodeURIComponent(key) + '=' + encodeURIComponent(value));
+      return memo;
+    },
+    []
+  ).join('&');
+}
+
+const MAX_SERIES = 20;
+export default class InfluxDatasource {
+  type: string;
+  urls: any;
+  username: string;
+  password: string;
+  name: string;
+  orgName: string;
+  database: any;
+  basicAuth: any;
+  withCredentials: any;
+  interval: any;
+  supportAnnotations: boolean;
+  supportMetrics: boolean;
+
+  /** @ngInject */
+  constructor(instanceSettings, private backendSrv, private templateSrv) {
+    this.type = 'influxdb-ifql';
+    this.urls = instanceSettings.url.split(',').map(url => url.trim());
```

Comment:
this multi urls support is not something other data sources support so not sure we should add this here, maybe can be done in a more generic way for many data sources.

Reply:
The existing influx datasource does:
https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/influxdb/datasource.ts#L26

https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/influxdb/datasource.ts#L229

I don't know if it is used much

Reply:
Yeah, it's doing round-robin load balancing. On the one hand we should make migration easy, but then again users need to change the address anyway. Plus they may not have the same number of hosts for IFQL. I'll remove this feature.
Also, it can be argued that load balancing falls not within the remit of Grafana.